packages/ember-engines/addon/components/link-to-component.js

```diff
@@ -1,13 +1,21 @@
-import LinkComponent from '@ember/routing/link-component';
-import { getOwner } from '@ember/application';
+import RoutingLinkComponent from '@ember/routing/link-component';
+import Application, { getOwner } from '@ember/application';
 import { computed, get, set } from '@ember/object';
 import { typeOf } from '@ember/utils';
 import { assert, deprecate } from '@ember/debug';
-import { macroCondition, dependencySatisfies } from '@embroider/macros';
+import { macroCondition, getOwnConfig, importSync, dependencySatisfies } from '@embroider/macros';
 
 let LinkTo;
+let LinkComponent;
 
-if (macroCondition(dependencySatisfies('ember-source', '>= 3.24.1'))) {
+if (macroCondition(dependencySatisfies('@ember/legacy-built-in-components', '*'))) {
+  let { LinkComponent: LegacyLinkComponent } = importSync('@ember/legacy-built-in-components');
```

Comment:
Thank you for doing that @BobrImperator. But I prefer the solution made by @SergeAstapov [here](https://github.com/ember-engines/ember-engines/pull/796) it's more composable 😄

Reply:
I'd prefer it too, but it's not working yet if I understand the description correctly.

We can rebase after/before it lands